jigglypuffRL/common/values.py

```diff
@@ -34,7 +34,39 @@ def __init__(self, state_dim, action_dim=None, val_type="V", hidden=(32, 32)):
         self.model = _get_val_model(mlp, val_type, state_dim, hidden, action_dim)
 
 
-value_registry = {"mlp": MlpValue}
+def CNNValue(BaseValue):
+    """
+    CNN Value
+    :param state_dim: (int) state dimension of environment
+    :param action_dim: (int) action dimension of environment
+    :param history_length: (int) length of history of states
+    :param val_type: (str) type of value function.
+        'V' for V(s), 'Qs' for Q(s), 'Qsa' for Q(s,a)
+    :param hidden: (tuple or list) sizes of hidden layers
+    :param disc: (bool) discrete action space?
+    """
+    def __init__(
+        self, state_dim, action_dim, history_length=4, val_type="Qs",
+        hidden=(256, 256), disc=True
```

Comment:
In `CNNValue` it would be preferable to have the `hidden` arg mean number of filters in every conv layer.  Lets stick the filter size to only 3 for now

Reply:
Actually lets stick to NatureCNN for a while. We can track an issue to allow for ease over creation of CNN networks after this PR is merged.

Reply:
I'm not quite sure I understand what you're saying. I haven't yet read about Impala CNN if that's what you mean by generalization. As for the `hidden` arg should I perhaps rename it to `fc_layers` or something like that?

Reply:
Yeah, I think just renaming should be ok for now.  Can you also separate `params` list into two? It would be better to just have the usual names like `kernels`, `strides`, etc